source/Calamari/Kubernetes/Commands/HelmUpgradeCommand.cs

```diff
@@ -138,6 +140,13 @@ IEnumerable<string> ExplicitlySpecifiedValuesFiles()
                 }
             }
         }
+        
+        IList<string> TemplateValuesFiles(RunningDeployment deployment)
+        {
+            //we pass null here for the log as we don't want to log the values files being included twice. They will be properly logged during the deployment
+            return HelmTemplateValueSourcesParser.ParseTemplateValuesFilesFromDependencies(deployment, fileSystem, null)
+                                                 .ToList();
```

Comment:
As the comment says, if we include the log, it does the `Included files...` log lines twice. Suppressing those log messages here then let's them show up later.

Reply:
This feels a little unexpected. Its not clear from either inside or outside the class that this is the intention of nulling a log object, which is usually a cross cutting concern.

Looking at the `HelmTemplateValueSourcesParser` class itself, I'm actually thinking it should perhapos be turned into a class, rather than a static method, passing the `ICalamariFileSystem fileSystem, ILog log` in the ctor. The result of this is that the actual method invocation can take parameters that describe the intention of the call. 

So it might take `Deployment deployment, bool logFoundFiles` or something to that effect. 

It might be a little overhead since its not mutating any state on the object, but it makes the intention a little clearer and less surprising.

Reply:
@zentron Would you be comfortable if I merge this, then refactor it in a separate PR?
Just so we can get this fix moving through to customers